### PR TITLE
Check CoinType is initialized in register

### PR DESCRIPTION
**Pull request by movekevin**
### Description
This prevents accounts from registering arbitrary types that are not valid Coins.

### Test Plan
Unit tests

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/coin.move [hunk 1]
```diff
@@ -497,6 +497,8 @@ module aptos_framework::coin {
     }
 
     public fun register<CoinType>(account: &signer) {
+        assert!(is_coin_initialized<CoinType>(), error::invalid_argument(ECOIN_INFO_NOT_PUBLISHED));
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/coin.move:500]: Doesn't this serialize registers on looking up if the coin info exists?

**Review by gregnazario — APPROVED** [review]

**Review by lightmark — APPROVED** [review]
It's funny...

File: aptos-move/framework/aptos-framework/sources/coin.move (hunk 1 above)
Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/coin.move:500]: just to be clear, we're going to add in a new read resource for each time we create an account or initialize a coin

I think that's probably why it was left out

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/coin.move:500]: This shouldn't a problem now with reduced gas

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/coin.move:500]: Based on the land-blocking forge test, there doesn't seem to be any execution degradation that affects P2P tx throughput (relatively to the test on another recent PR)

**Comment by davidiw** [comment]
Let's not leave fully accepted PRs open for multiple months. Closing out, but feel free to re-open.

**Comment by davidiw** [comment]
Note, this will require a bit more plumbing to actually make work.

**Pull request closed without merging**